pkg/util/net/ip/ip.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+package ip
+
+import (
+	"io/ioutil"
+	"net"
+
+	netutils "k8s.io/utils/net"
+)
+
+type readFileFunc func(filename string) ([]byte, error)
+
+var GetLoopbackAddress = getLoopbackAddress
+var readFile = ioutil.ReadFile
```

Comment:
I compromized for the `GetLoopbackAddress` global because it is hard to do it right with the current production code, requiring a lot of refactoring.

But here we have new code, which can be easily built right with tests without global hacks.
Here is the option you could use to have this sorted out without globals:
```
func GetIPZeroAddress() string {
	return getIPZeroAddress(isIPv6Disabled(disableIPv6Path))
}
func getIPZeroAddress(disableIPv6Disabled bool) string {
	if disableIPv6Disabled {
		return net.IPv4zero.String()
	}
	return net.IPv6zero.String()
}
```

It separates the io part for the logic part (although it can be done even more, but this seems enough).
Allowing you to test `getIPZeroAddress`.

Reply:
see please that i did not override the var from outside directly, it isnt exported even
i added function that do it.
and this method that i did even can remove the need of that other global that already exists.

The code you suggested indeed separate the io from the logic, 
but the code becomes, imo, less readable and i feel that the production code should be readable and easier as it can,

but i will change if you think otherwise, this is my opinion at least.